src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -650,7 +698,8 @@ class FromClauseSegment(BaseSegment):
                 Ref('CrossKeywordSegment'),
```

Comment:
Can you replace all lines 698 until 702 with the new JoinClauseSegment?

Reply:
potentially yes. There *might* be a performance impact of that but I'll investigate...